### PR TITLE
Dao script

### PR DESCRIPTION
**Pull request by dev-matan-tsuberi**

### CODE REVIEW

**Comment by dev-matan-tsuberi** [comment]
[Populate the DAO script](https://trello.com/c/SJ98eY7N/504-populate-the-dao-script)

File: package.json [hunk 1]
```diff
@@ -144,12 +148,14 @@
     "sass-resources-loader": "^1.3.0",
     "source-map-loader": "^0.2.1",
     "ts-jest": "^22.4.1",
+    "ts-node": "^5.0.1",
```
Comment (tibetsprague) [inline package.json]: why is this now a devDependency? its already a regular dependency

File: scripts/example.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
```
Comment (tibetsprague) [inline scripts/example.json]: i think the name should be something more descriptive than example.json. exampleDaoData.json? also maybe this should go in the test directory?

File: scripts/populateDAO.ts [hunk 3]
```diff
@@ -0,0 +1,335 @@
+import * as yargs from 'yargs';
```
Comment (tibetsprague) [inline scripts/populateDAO.ts]: lets alphabetize our includes

File: scripts/populateDAO.ts [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+import chalk from 'chalk';
+import * as moment from 'moment';
+import * as fs from 'fs';
+import * as Arc from '@daostack/arc.js';
+import { VoteOptions, DAO } from '@daostack/arc.js';
+import Util from '../src/lib/util';
+import axios from "axios";
+import { ContributionRewardWrapper, GenesisProtocolWrapper } from '@daostack/arc.js';
```
Comment (tibetsprague) [inline scripts/populateDAO.ts]: also put the arc.js stuff together

File: scripts/populateDAO.ts [hunk 5]
```diff
@@ -0,0 +1,335 @@
+import * as yargs from 'yargs';
+import chalk from 'chalk';
+import * as moment from 'moment';
+import * as fs from 'fs';
+import * as Arc from '@daostack/arc.js';
+import { VoteOptions, DAO } from '@daostack/arc.js';
+import Util from '../src/lib/util';
```
Comment (tibetsprague) [inline scripts/populateDAO.ts]: and our code after third party vendors

File: scripts/populateDAO.ts [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+const Web3 = require("web3");
+import promisify = require("es6-promisify");
+// tslint:disable-next-line:no-var-requires
+const NonceTrackerSubprovider = require("web3-provider-engine/subproviders/nonce-tracker");
+
+// tslint:disable-next-line:no-var-requires
+require('dotenv').config();
+
+enum Level {
+    Success = 'success',
```
Comment (tibetsprague) [inline scripts/populateDAO.ts]: looks like you may have used tabs in this file? lets stick to two spaces everywhere

File: scripts/populateDAO.ts [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+interface Opts extends CreateDAOOpts {
+    steps: Step[];
+    api: string;
+    network: string;
+    logfile: string;
+    mnemonic: string;
+}
+
+async function main(options: Opts) {
+    const mapping: {[id: number]: string} = {};
```
Comment (tibetsprague) [inline scripts/populateDAO.ts]: mapping what? better name please

File: scripts/populateDAO.ts [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+                alias: 'l',
+                describe: 'path to a log file to save results',
+                type: 'string'
+            }
+        )
+        .option(
+            'api', {
+                alias: 'a',
+                describe: 'api root url',
+                default: 'https://daostack-alchemy.herokuapp.com',
```
Comment (tibetsprague) [inline scripts/populateDAO.ts]: hmm, i think we should not default to production server, lets default to daostack-alchemy-server-stage

Comment (tibetsprague) [reply scripts/populateDAO.ts]: though really we just need to always use the right server for the network used, im going to pull that ticket in

File: package.json (hunk 1 above)
Comment (dev-matan-tsuberi) [reply package.json]: does it need to be a regular dependency? its should not be bundled with the app.

Comment (tibetsprague) [reply package.json]: we use it to run scripts on Heroku

Comment (dev-matan-tsuberi) [reply package.json]: doesn't it `npm install` it and runs the command inside the dir?

**Pull request merged**